### CAPSTONE PROJECT

### CASE STUDY: CYCLISTIC BIKE SHARING

#### BUSINESS TASK:
What is the problem?

1.How do annual members and casual riders use Cyclistic bikes differently?

2.Why would casual riders buy Cyclistic annual memberships?

3.How can Cyclists use digital media to influence casual riders to become members?

4.How can your insights drive business decisions?

If we know how casual riders differ from annual members, we can develop a strategy to target casual riders and convert them into annual members.


#### DATASET:
The data is on an AWS server where it is easily downloadable and named correctly. I downloaded the previous 11 months data for the year 2022, and stored it locally for the next steps in the analysis processes. The data is current, cited, original but not reliable. There are some null values and outliers in the data.

Note that data-privacy issues prohibits me from using riders’ personally identifiable information. This means that you won’t be able to connect pass purchases to credit card numbers to determine if casual riders live in the Cyclistic service area or if they have purchased multiple single passes.

The files are saved in comma-seperated values (CSV).

Python is used to analyze this data

Link to previous 12 months of Cyclistic Trip Data: https://divvy-tripdata.s3.amazonaws.com/index.html  

In [3]:
# importing libraries
import pandas as pd
import numpy as np
import seaborn as sns
import datetime 
import matplotlib.pyplot as plt

In [5]:
# Importing the downloaded datasets
data_1 = pd.read_csv('01-january-2022.csv')
data_2 = pd.read_csv('02-february-2022.csv')
data_3 = pd.read_csv('03-march-2022.csv')
data_4 = pd.read_csv('04-april-2022.csv')
data_5 = pd.read_csv('05-may-2022.csv')
data_6 = pd.read_csv('06-june-2022.csv')
data_7 = pd.read_csv('07-july-2022.csv')
data_8 = pd.read_csv('08-august-2022.csv')
data_9 = pd.read_csv('09-september-2022.csv')
data_10 = pd.read_csv('10-october-2022.csv')
data_11 = pd.read_csv('11-november-2022.csv')

In [6]:
data_1.head()

ride_id  rideable_type           started_at             ended_at  \
0  C2F7DD78E82EC875  electric_bike  2022-01-13 11:59:47  2022-01-13 12:02:44   
1  A6CF8980A652D272  electric_bike  2022-01-10 08:41:56  2022-01-10 08:46:17   
2  BD0F91DFF741C66D   classic_bike  2022-01-25 04:53:40  2022-01-25 04:58:01   
3  CBB80ED419105406   classic_bike  2022-01-04 00:18:04  2022-01-04 00:33:00   
4  DDC963BFDDA51EEA   classic_bike  2022-01-20 01:31:10  2022-01-20 01:37:12   

              start_station_name start_station_id  \
0       Glenwood Ave & Touhy Ave              525   
1       Glenwood Ave & Touhy Ave              525   
2  Sheffield Ave & Fullerton Ave     TA1306000016   
3       Clark St & Bryn Mawr Ave     KA1504000151   
4    Michigan Ave & Jackson Blvd     TA1309000002   

                end_station_name end_station_id  start_lat  start_lng  \
0           Clark St & Touhy Ave         RP-007  42.012800 -87.665906   
1           Clark St & Touhy Ave         RP-007  42.012763 -87.665967   
2  Greenview Ave & Fullerton Ave   TA1307000001  41.925602 -87.653708   
3      Paulina St & Montrose Ave   TA1309000021  41.983593 -87.669154   
4         State St & Randolph St   TA1305000029  41.877850 -87.624080   

     end_lat    end_lng member_casual  
0  42.012560 -87.674367        casual  
1  42.012560 -87.674367        casual  
2  41.925330 -87.665800        member  
3  41.961507 -87.671387        casual  
4  41.884621 -87.627834        member

In [7]:
data_5.head()

ride_id rideable_type           started_at             ended_at  \
0  EC2DE40644C6B0F4  classic_bike  2022-05-23 23:06:58  2022-05-23 23:40:19   
1  1C31AD03897EE385  classic_bike  2022-05-11 08:53:28  2022-05-11 09:31:22   
2  1542FBEC830415CF  classic_bike  2022-05-26 18:36:28  2022-05-26 18:58:18   
3  6FF59852924528F8  classic_bike  2022-05-10 07:30:07  2022-05-10 07:38:49   
4  483C52CAAE12E3AC  classic_bike  2022-05-10 17:31:56  2022-05-10 17:36:57   

                  start_station_name start_station_id  \
0             Wabash Ave & Grand Ave     TA1307000117   
1  DuSable Lake Shore Dr & Monroe St            13300   
2            Clinton St & Madison St     TA1305000032   
3            Clinton St & Madison St     TA1305000032   
4            Clinton St & Madison St     TA1305000032   

              end_station_name end_station_id  start_lat  start_lng  \
0       Halsted St & Roscoe St   TA1309000025  41.891466 -87.626761   
1  Field Blvd & South Water St          15534  41.880958 -87.616743   
2      Wood St & Milwaukee Ave          13221  41.882242 -87.641066   
3       Clark St & Randolph St   TA1305000030  41.882242 -87.641066   
4          Morgan St & Lake St   TA1306000015  41.882242 -87.641066   

     end_lat    end_lng member_casual  
0  41.943670 -87.648950        member  
1  41.886349 -87.617517        member  
2  41.907655 -87.672552        member  
3  41.884576 -87.631890        member  
4  41.885779 -87.651025        member

In [19]:
# checking the info of each monthly data, and checking for any null values
#It was discovered there was no null values in all 11 months data for 2022
data_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103770 entries, 0 to 103769
Data columns (total 13 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   ride_id             103770 non-null  object 
 1   rideable_type       103770 non-null  object 
 2   started_at          103770 non-null  object 
 3   ended_at            103770 non-null  object 
 4   start_station_name  87510 non-null   object 
 5   start_station_id    87510 non-null   object 
 6   end_station_name    85843 non-null   object 
 7   end_station_id      85843 non-null   object 
 8   start_lat           103770 non-null  float64
 9   start_lng           103770 non-null  float64
 10  end_lat             103684 non-null  float64
 11  end_lng             103684 non-null  float64
 12  member_casual       103770 non-null  object 
dtypes: float64(4), object(9)
memory usage: 10.3+ MB


In [22]:
# Merging the data
bike_data = pd.concat([data_1, data_2, data_3, data_4, data_5, data_6, data_7, data_8, data_9, data_10, data_11])
bike_data

ride_id  rideable_type           started_at  \
0       C2F7DD78E82EC875  electric_bike  2022-01-13 11:59:47   
1       A6CF8980A652D272  electric_bike  2022-01-10 08:41:56   
2       BD0F91DFF741C66D   classic_bike  2022-01-25 04:53:40   
3       CBB80ED419105406   classic_bike  2022-01-04 00:18:04   
4       DDC963BFDDA51EEA   classic_bike  2022-01-20 01:31:10   
...                  ...            ...                  ...   
337730  C349E243A9BAA6F7  electric_bike  2022-11-25 11:19:52   
337731  B0B4E85DA43A9194   classic_bike  2022-11-22 16:57:53   
337732  8D148DD47B59530B   classic_bike  2022-11-06 13:04:05   
337733  0D1170BA18FD33D1   classic_bike  2022-11-06 09:41:29   
337734  09B20DC75B5EA1E0  electric_bike  2022-11-26 11:59:28   

                   ended_at             start_station_name start_station_id  \
0       2022-01-13 12:02:44       Glenwood Ave & Touhy Ave              525   
1       2022-01-10 08:46:17       Glenwood Ave & Touhy Ave              525   
2       2022-01-25 04:58:01  Sheffield Ave & Fullerton Ave     TA1306000016   
3       2022-01-04 00:33:00       Clark St & Bryn Mawr Ave     KA1504000151   
4       2022-01-20 01:37:12    Michigan Ave & Jackson Blvd     TA1309000002   
...                     ...                            ...              ...   
337730  2022-11-25 11:31:50         Wabash Ave & Grand Ave     TA1307000117   
337731  2022-11-22 17:31:29     Franklin St & Jackson Blvd     TA1305000025   
337732  2022-11-06 13:13:33  Michigan Ave & Ida B Wells Dr     TA1305000010   
337733  2022-11-06 15:17:17           Halsted St & 18th St            13099   
337734  2022-11-26 12:31:04  Michigan Ave & Ida B Wells Dr     TA1305000010   

                      end_station_name end_station_id  start_lat  start_lng  \
0                 Clark St & Touhy Ave         RP-007  42.012800 -87.665906   
1                 Clark St & Touhy Ave         RP-007  42.012763 -87.665967   
2        Greenview Ave & Fullerton Ave   TA1307000001  41.925602 -87.653708   
3            Paulina St & Montrose Ave   TA1309000021  41.983593 -87.669154   
4               State St & Randolph St   TA1305000029  41.877850 -87.624080   
...                                ...            ...        ...        ...   
337730                             NaN            NaN  41.891836 -87.626806   
337731  Sheffield Ave & Wrightwood Ave   TA1309000023  41.877708 -87.635321   
337732                  Shedd Aquarium          15544  41.876243 -87.624426   
337733                  Shedd Aquarium          15544  41.857506 -87.645991   
337734                  Shedd Aquarium          15544  41.876272 -87.624576   

          end_lat    end_lng member_casual  
0       42.012560 -87.674367        casual  
1       42.012560 -87.674367        casual  
2       41.925330 -87.665800        member  
3       41.961507 -87.671387        casual  
4       41.884621 -87.627834        member  
...           ...        ...           ...  
337730  41.890000 -87.620000        member  
337731  41.928712 -87.653833        casual  
337732  41.867226 -87.615355        casual  
337733  41.867226 -87.615355        casual  
337734  41.867226 -87.615355        casual  

[5485911 rows x 13 columns]

In [23]:
bike_data.head()

ride_id  rideable_type           started_at             ended_at  \
0  C2F7DD78E82EC875  electric_bike  2022-01-13 11:59:47  2022-01-13 12:02:44   
1  A6CF8980A652D272  electric_bike  2022-01-10 08:41:56  2022-01-10 08:46:17   
2  BD0F91DFF741C66D   classic_bike  2022-01-25 04:53:40  2022-01-25 04:58:01   
3  CBB80ED419105406   classic_bike  2022-01-04 00:18:04  2022-01-04 00:33:00   
4  DDC963BFDDA51EEA   classic_bike  2022-01-20 01:31:10  2022-01-20 01:37:12   

              start_station_name start_station_id  \
0       Glenwood Ave & Touhy Ave              525   
1       Glenwood Ave & Touhy Ave              525   
2  Sheffield Ave & Fullerton Ave     TA1306000016   
3       Clark St & Bryn Mawr Ave     KA1504000151   
4    Michigan Ave & Jackson Blvd     TA1309000002   

                end_station_name end_station_id  start_lat  start_lng  \
0           Clark St & Touhy Ave         RP-007  42.012800 -87.665906   
1           Clark St & Touhy Ave         RP-007  42.012763 -87.665967   
2  Greenview Ave & Fullerton Ave   TA1307000001  41.925602 -87.653708   
3      Paulina St & Montrose Ave   TA1309000021  41.983593 -87.669154   
4         State St & Randolph St   TA1305000029  41.877850 -87.624080   

     end_lat    end_lng member_casual  
0  42.012560 -87.674367        casual  
1  42.012560 -87.674367        casual  
2  41.925330 -87.665800        member  
3  41.961507 -87.671387        casual  
4  41.884621 -87.627834        member

In [24]:
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5485911 entries, 0 to 337734
Data columns (total 13 columns):
 #   Column              Dtype  
---  ------              -----  
 0   ride_id             object 
 1   rideable_type       object 
 2   started_at          object 
 3   ended_at            object 
 4   start_station_name  object 
 5   start_station_id    object 
 6   end_station_name    object 
 7   end_station_id      object 
 8   start_lat           float64
 9   start_lng           float64
 10  end_lat             float64
 11  end_lng             float64
 12  member_casual       object 
dtypes: float64(4), object(9)
memory usage: 586.0+ MB


The merged data set for the 11 months (January to November, 2022) conists of 5,485,911 rows and 13 columns in total.

In [26]:
# Creating a backup of the original merged data
bike_data.to_csv('bike_data1')

In [29]:
# changing ride_type and member columns to category data types
bike_data['rideable_type'] = bike_data.rideable_type.astype('category')
bike_data['member_casual'] = bike_data.member_casual.astype('category')

In [30]:
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5485911 entries, 0 to 337734
Data columns (total 13 columns):
 #   Column              Dtype   
---  ------              -----   
 0   ride_id             object  
 1   rideable_type       category
 2   started_at          object  
 3   ended_at            object  
 4   start_station_name  object  
 5   start_station_id    object  
 6   end_station_name    object  
 7   end_station_id      object  
 8   start_lat           float64 
 9   start_lng           float64 
 10  end_lat             float64 
 11  end_lng             float64 
 12  member_casual       category
dtypes: category(2), float64(4), object(7)
memory usage: 512.7+ MB


In [32]:
# dropping the lat and lng columns from the datasets
bike_data.drop(['start_lat', 'start_lng', 'end_lat', 'end_lng'], axis = 1, inplace = True)

In [34]:
bike_data.info
bike_data.head()

ride_id  rideable_type           started_at             ended_at  \
0  C2F7DD78E82EC875  electric_bike  2022-01-13 11:59:47  2022-01-13 12:02:44   
1  A6CF8980A652D272  electric_bike  2022-01-10 08:41:56  2022-01-10 08:46:17   
2  BD0F91DFF741C66D   classic_bike  2022-01-25 04:53:40  2022-01-25 04:58:01   
3  CBB80ED419105406   classic_bike  2022-01-04 00:18:04  2022-01-04 00:33:00   
4  DDC963BFDDA51EEA   classic_bike  2022-01-20 01:31:10  2022-01-20 01:37:12   

              start_station_name start_station_id  \
0       Glenwood Ave & Touhy Ave              525   
1       Glenwood Ave & Touhy Ave              525   
2  Sheffield Ave & Fullerton Ave     TA1306000016   
3       Clark St & Bryn Mawr Ave     KA1504000151   
4    Michigan Ave & Jackson Blvd     TA1309000002   

                end_station_name end_station_id member_casual  
0           Clark St & Touhy Ave         RP-007        casual  
1           Clark St & Touhy Ave         RP-007        casual  
2  Greenview Ave & Fullerton Ave   TA1307000001        member  
3      Paulina St & Montrose Ave   TA1309000021        casual  
4         State St & Randolph St   TA1305000029        member

In [36]:
# changing start time and end time to a datetime datatype
bike_data['started_at'] = pd.to_datetime(bike_data.started_at)
bike_data['ended_at'] = pd.to_datetime(bike_data.ended_at)

In [38]:
bike_data
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5485911 entries, 0 to 337734
Data columns (total 9 columns):
 #   Column              Dtype         
---  ------              -----         
 0   ride_id             object        
 1   rideable_type       category      
 2   started_at          datetime64[ns]
 3   ended_at            datetime64[ns]
 4   start_station_name  object        
 5   start_station_id    object        
 6   end_station_name    object        
 7   end_station_id      object        
 8   member_casual       category      
dtypes: category(2), datetime64[ns](2), object(5)
memory usage: 345.3+ MB


In [41]:
# finding the missing values
# start station nae, the id, end station name and the id all has over 800,000 missing values
bike_data.isnull().sum()

ride_id                    0
rideable_type              0
started_at                 0
ended_at                   0
start_station_name    803781
start_station_id      803781
end_station_name      861584
end_station_id        861584
member_casual              0
dtype: int64

In [43]:
# adding a new column of the ride_length, (ended_at - started_at) to calculate the time travelled
bike_data['ride_length'] = bike_data['ended_at'] - bike_data['started_at']
bike_data['ride_length']

0        0 days 00:02:57
1        0 days 00:04:21
2        0 days 00:04:21
3        0 days 00:14:56
4        0 days 00:06:02
               ...      
337730   0 days 00:11:58
337731   0 days 00:33:36
337732   0 days 00:09:28
337733   0 days 05:35:48
337734   0 days 00:31:36
Name: ride_length, Length: 5485911, dtype: timedelta64[ns]

In [44]:
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5485911 entries, 0 to 337734
Data columns (total 10 columns):
 #   Column              Dtype          
---  ------              -----          
 0   ride_id             object         
 1   rideable_type       category       
 2   started_at          datetime64[ns] 
 3   ended_at            datetime64[ns] 
 4   start_station_name  object         
 5   start_station_id    object         
 6   end_station_name    object         
 7   end_station_id      object         
 8   member_casual       category       
 9   ride_length         timedelta64[ns]
dtypes: category(2), datetime64[ns](2), object(5), timedelta64[ns](1)
memory usage: 387.2+ MB


In [46]:
# splitting and extracting the date from the 'started_at' column
bike_data['start_date'] = bike_data.started_at.dt.date
bike_data.start_date.head()

0    2022-01-13
1    2022-01-10
2    2022-01-25
3    2022-01-04
4    2022-01-20
Name: start_date, dtype: object

In [47]:
# converting 'day_of_week' into a datetime dtype
bike_data['start_date'] = pd.to_datetime(bike_data.start_date)
bike_data.dtypes

ride_id                        object
rideable_type                category
started_at             datetime64[ns]
ended_at               datetime64[ns]
start_station_name             object
start_station_id               object
end_station_name               object
end_station_id                 object
member_casual                category
ride_length           timedelta64[ns]
start_date             datetime64[ns]
dtype: object

In [49]:
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5485911 entries, 0 to 337734
Data columns (total 11 columns):
 #   Column              Dtype          
---  ------              -----          
 0   ride_id             object         
 1   rideable_type       category       
 2   started_at          datetime64[ns] 
 3   ended_at            datetime64[ns] 
 4   start_station_name  object         
 5   start_station_id    object         
 6   end_station_name    object         
 7   end_station_id      object         
 8   member_casual       category       
 9   ride_length         timedelta64[ns]
 10  start_date          datetime64[ns] 
dtypes: category(2), datetime64[ns](3), object(5), timedelta64[ns](1)
memory usage: 429.0+ MB


In [51]:
# finding the weekday of the date
bike_data['day_name'] = bike_data['start_date'].dt.day_name()
bike_data['day_name'].unique()

array(['Thursday', 'Monday', 'Tuesday', 'Sunday', 'Saturday', 'Friday',
       'Wednesday'], dtype=object)

In [52]:
# changing 'day_name' to category
bike_data['day_name'] = bike_data.day_name.astype('category')
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5485911 entries, 0 to 337734
Data columns (total 12 columns):
 #   Column              Dtype          
---  ------              -----          
 0   ride_id             object         
 1   rideable_type       category       
 2   started_at          datetime64[ns] 
 3   ended_at            datetime64[ns] 
 4   start_station_name  object         
 5   start_station_id    object         
 6   end_station_name    object         
 7   end_station_id      object         
 8   member_casual       category       
 9   ride_length         timedelta64[ns]
 10  start_date          datetime64[ns] 
 11  day_name            category       
dtypes: category(3), datetime64[ns](3), object(5), timedelta64[ns](1)
memory usage: 434.2+ MB


In [53]:
bike_data['year'] = pd.DatetimeIndex(bike_data['start_date']).year
bike_data['month'] = pd.DatetimeIndex(bike_data['start_date']).month

In [54]:
bike_data.head()

ride_id  rideable_type          started_at            ended_at  \
0  C2F7DD78E82EC875  electric_bike 2022-01-13 11:59:47 2022-01-13 12:02:44   
1  A6CF8980A652D272  electric_bike 2022-01-10 08:41:56 2022-01-10 08:46:17   
2  BD0F91DFF741C66D   classic_bike 2022-01-25 04:53:40 2022-01-25 04:58:01   
3  CBB80ED419105406   classic_bike 2022-01-04 00:18:04 2022-01-04 00:33:00   
4  DDC963BFDDA51EEA   classic_bike 2022-01-20 01:31:10 2022-01-20 01:37:12   

              start_station_name start_station_id  \
0       Glenwood Ave & Touhy Ave              525   
1       Glenwood Ave & Touhy Ave              525   
2  Sheffield Ave & Fullerton Ave     TA1306000016   
3       Clark St & Bryn Mawr Ave     KA1504000151   
4    Michigan Ave & Jackson Blvd     TA1309000002   

                end_station_name end_station_id member_casual     ride_length  \
0           Clark St & Touhy Ave         RP-007        casual 0 days 00:02:57   
1           Clark St & Touhy Ave         RP-007        casual 0 days 00:04:21   
2  Greenview Ave & Fullerton Ave   TA1307000001        member 0 days 00:04:21   
3      Paulina St & Montrose Ave   TA1309000021        casual 0 days 00:14:56   
4         State St & Randolph St   TA1305000029        member 0 days 00:06:02   

  start_date  day_name  year  month  
0 2022-01-13  Thursday  2022      1  
1 2022-01-10    Monday  2022      1  
2 2022-01-25   Tuesday  2022      1  
3 2022-01-04   Tuesday  2022      1  
4 2022-01-20  Thursday  2022      1

In [55]:
# sort the dataframe by ascending using 'started_at'
bike_data.sort_values(by = ['started_at'], inplace = True, ascending = True)

In [56]:
bike_data.head()

ride_id  rideable_type          started_at  \
73479  98D355D9A9852BE9   classic_bike 2022-01-01 00:00:05   
68510  04706CA7F5BD25EE  electric_bike 2022-01-01 00:01:00   
64490  42178E850B92597A  electric_bike 2022-01-01 00:01:16   
73485  6B93C46E8F5B114C   classic_bike 2022-01-01 00:02:14   
73478  466943353EAC8022   classic_bike 2022-01-01 00:02:35   

                 ended_at         start_station_name start_station_id  \
73479 2022-01-01 00:01:48      Michigan Ave & 8th St              623   
68510 2022-01-01 00:04:39    Broadway & Waveland Ave            13325   
64490 2022-01-01 00:32:14  Clark St & Ida B Wells Dr     TA1305000009   
73485 2022-01-01 00:31:07      Michigan Ave & 8th St              623   
73478 2022-01-01 00:31:04      Michigan Ave & 8th St              623   

                end_station_name end_station_id member_casual     ride_length  \
73479      Michigan Ave & 8th St            623        casual 0 days 00:01:43   
68510       Broadway & Barry Ave          13137        casual 0 days 00:03:39   
64490  Clark St & Ida B Wells Dr   TA1305000009        casual 0 days 00:30:58   
73485      Michigan Ave & 8th St            623        casual 0 days 00:28:53   
73478      Michigan Ave & 8th St            623        casual 0 days 00:28:29   

      start_date  day_name  year  month  
73479 2022-01-01  Saturday  2022      1  
68510 2022-01-01  Saturday  2022      1  
64490 2022-01-01  Saturday  2022      1  
73485 2022-01-01  Saturday  2022      1  
73478 2022-01-01  Saturday  2022      1

In [59]:
bike_data.isnull().sum()

ride_id                    0
rideable_type              0
started_at                 0
ended_at                   0
start_station_name    803781
start_station_id      803781
end_station_name      861584
end_station_id        861584
member_casual              0
ride_length                0
start_date                 0
day_name                   0
year                       0
month                      0
dtype: int64

In [61]:
# drop all rows with NAN values 
bike_data.dropna()
bike_data.dropna(inplace = True)

In [64]:
# to save the cleaned file
bike_data.to_csv('bike_data_cleaned.csv')

In [63]:
# print cleaned dataset
bike_data.head()

ride_id  rideable_type          started_at  \
73479  98D355D9A9852BE9   classic_bike 2022-01-01 00:00:05   
68510  04706CA7F5BD25EE  electric_bike 2022-01-01 00:01:00   
64490  42178E850B92597A  electric_bike 2022-01-01 00:01:16   
73485  6B93C46E8F5B114C   classic_bike 2022-01-01 00:02:14   
73478  466943353EAC8022   classic_bike 2022-01-01 00:02:35   

                 ended_at         start_station_name start_station_id  \
73479 2022-01-01 00:01:48      Michigan Ave & 8th St              623   
68510 2022-01-01 00:04:39    Broadway & Waveland Ave            13325   
64490 2022-01-01 00:32:14  Clark St & Ida B Wells Dr     TA1305000009   
73485 2022-01-01 00:31:07      Michigan Ave & 8th St              623   
73478 2022-01-01 00:31:04      Michigan Ave & 8th St              623   

                end_station_name end_station_id member_casual     ride_length  \
73479      Michigan Ave & 8th St            623        casual 0 days 00:01:43   
68510       Broadway & Barry Ave          13137        casual 0 days 00:03:39   
64490  Clark St & Ida B Wells Dr   TA1305000009        casual 0 days 00:30:58   
73485      Michigan Ave & 8th St            623        casual 0 days 00:28:53   
73478      Michigan Ave & 8th St            623        casual 0 days 00:28:29   

      start_date  day_name  year  month  
73479 2022-01-01  Saturday  2022      1  
68510 2022-01-01  Saturday  2022      1  
64490 2022-01-01  Saturday  2022      1  
73485 2022-01-01  Saturday  2022      1  
73478 2022-01-01  Saturday  2022      1

In [67]:
# checking if the cleaned data has null values
bike_data.isnull().sum()

ride_id               0
rideable_type         0
started_at            0
ended_at              0
start_station_name    0
start_station_id      0
end_station_name      0
end_station_id        0
member_casual         0
ride_length           0
start_date            0
day_name              0
year                  0
month                 0
dtype: int64

In [70]:
# checking for any duplicated values
bike_data.duplicated().any()

False

In [71]:
# getting info of the cleaned data
bike_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4233957 entries, 73479 to 337271
Data columns (total 14 columns):
 #   Column              Dtype          
---  ------              -----          
 0   ride_id             object         
 1   rideable_type       category       
 2   started_at          datetime64[ns] 
 3   ended_at            datetime64[ns] 
 4   start_station_name  object         
 5   start_station_id    object         
 6   end_station_name    object         
 7   end_station_id      object         
 8   member_casual       category       
 9   ride_length         timedelta64[ns]
 10  start_date          datetime64[ns] 
 11  day_name            category       
 12  year                int64          
 13  month               int64          
dtypes: category(3), datetime64[ns](3), int64(2), object(5), timedelta64[ns](1)
memory usage: 399.7+ MB
